test/integration/integration_test.cc

```diff
@@ -92,6 +92,39 @@ TEST_P(IntegrationTest, PerWorkerStatsAndBalancing) {
   check_listener_stats(0, 1);
 }
 
+// Validates that the drain actually drains the listeners.
+TEST_P(IntegrationTest, AdminDrainDrainsListeners) {
+  initialize();
+
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+  Http::TestHeaderMapImpl request_headers{{":method", "HEAD"},
+                                          {":path", "/test/long/url"},
+                                          {":scheme", "http"},
+                                          {":authority", "host"}};
+  IntegrationStreamDecoderPtr response = codec_client_->makeHeaderOnlyRequest(request_headers);
+  waitForNextUpstreamRequest(0);
+  fake_upstreams_[0]->set_allow_unexpected_disconnects(true);
+
+  upstream_request_->encodeHeaders(default_response_headers_, false);
+
+  // Invoke drain listeners endpoint and validate that the listener is actually draining.
```

Comment:
How have you verified it is draining?

Reply:
it validates that after the listener drain is invoked, we can send some data. The drain part is validated in the listener_manager_impl_test by expecting on stopListener call on worker